src/loci/formats/services/JZarrServiceImpl.java

```diff
@@ -99,51 +103,27 @@ public void open(String id, ZarrArray array) {
   }
   
   public Map<String, Object> getGroupAttr(String path) throws IOException, FormatException {
-    ZarrGroup group = null;
-    if (!path.toLowerCase().contains("s3:")) {
-      group = ZarrGroup.open(path);
-    }
-    else {
-      LOGGER.warn("S3 access currently not supported");
-      return null;
-    }
-    return group.getAttributes();
+    return getGroup(path).getAttributes();
   }
 
   public Map<String, Object> getArrayAttr(String path) throws IOException, FormatException {
     ZarrArray array = null;
-    if (!path.toLowerCase().contains("s3:")) {
+    if (s3fs == null) {
       array = ZarrArray.open(path);
     }
     else {
-      LOGGER.warn("S3 access currently not supported");
-      return null;
+      s3fs.updateRoot(getZarrRoot(s3fs.getRoot()) + stripZarrRoot(path));
+      array = ZarrArray.open(s3fs);
     }
-    return array.getAttributes();
+    return getArray(path).getAttributes();
```

Comment:
It looks like the `array` variable is no longer used, so lines 110-117 could be removed since they duplicate the call to `getArray`?

Reply:
Yeah that should have been removed, latest commit corrects this